sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -134,12 +159,15 @@ public Mono<Void> updateCheckpoint(Checkpoint checkpoint) {
         String prefix = prefixBuilder(checkpoint.getFullyQualifiedNamespace(), checkpoint.getEventHubName(), checkpoint.getConsumerGroup());
         String key = keyBuilder(prefix, checkpoint.getPartitionId());
         try (Jedis jedis = jedisPool.getResource()) {
-            //Case 1: new checkpoint
             if (!jedis.exists(prefix) || !jedis.exists(key)) {
+                //Case 1: new checkpoint
                 jedis.sadd(prefix, key);
                 jedis.hset(key, CHECKPOINT, new String(DEFAULT_SERIALIZER.serializeToBytes(checkpoint), StandardCharsets.UTF_8));
+            } else {
+                //Case 2: checkpoint already exists in Redis cache
+                jedis.hset(key, CHECKPOINT, new String(DEFAULT_SERIALIZER.serializeToBytes(checkpoint), StandardCharsets.UTF_8));
```

Comment:
I've noticed this pattern of serializing to bytes, then turning it back into a string. There is an overload that takes bytes[]. Is there a reason to use string? using string forces us to go over the data twice.... object -> bytes -> reads bytes back into string.

Reply:
I went ahead and changed the implementation to only go from object <-> bytes.